code/LeftAndMain.php

```diff
@@ -1584,6 +1585,7 @@ public function BatchActionsForm()
             $actionsMap[$action->Link] = $action->Title;
         }
 
+        $cmsMain = singleton(CMSMain::class);
```

Comment:
CMSMain may not exist (it's perfectly valid to use `silverstripe/admin` in a project without `silverstripe/cms`).
Use the `updateBatchActionsForm()` extension hook in the cms module instead.

Reply:
Yeah I thought that as well, but then I noticed that the LeftAndMain class already imports the `SilverStripeNavigator` class from the CMS module on [line 9](https://github.com/silverstripe/silverstripe-admin/blob/1/code/LeftAndMain.php#L9)
```
use SilverStripe\CMS\Controllers\SilverStripeNavigator;
```

But if you'd rather use an extension, I can totally do that. In fact that is how I'm implementing it in my own installations. Oh but what should I call the `LeftAndMainExtension` subclass and file? or is there an existing extension I should put it in?

Reply:
Yes please. The fact that it's using `SilverStripeNavigator` is already something we're going to have to unpick - it shouldn't do that.

> what should I call the LeftAndMainExtension subclass and file?

Anything you think suitably describes the intention of the class. Maybe `LeftAndMainBatchActionsExtension` or something along those lines.
It also doesn't have to subclass `LeftAndMainExtension` specifically - there are no methods in that class that are useful here. May as well just subclass `Extension`.

Reply:
> Yes please. The fact that it's using SilverStripeNavigator is already something we're going to have to unpick - it shouldn't do that.

Oh yeah then we definitely don't want to add more dependence on the CMS module.

> Maybe LeftAndMainBatchActionsExtension or something along those lines.

Totally makes sense. I'll go ahead and create that as a Extension subclass and add it to my other PR. Or should it be a separate PR? Theoretically it shouldn't effect the impact of the first PR (in CMS) if we add it to that, as even though the LiteralField will be added to the BatchActionsForm, it won't be displayed due to the `display:none;`

Reply:
> add it to my other PR. Or should it be a separate PR?

Add it to the PR you already created - it's part of the same work of making batch actions parameters usable.

Reply:
Should the `LeftAndMainBatchActionsExtension` be placed in the base of the `code/` folder in the CMS module, or in a sub-folder (like `code/Extensions` or something)?

Reply:
Maybe put it next to `LeftAndMainPageIconsExtension` (in `code/Controllers/`)